src/reducers/AppReducer.js

```diff
@@ -0,0 +1,250 @@
+import globalTypes from "../GlobalTypes";
+import { v4 as uuidv4 } from "uuid";
+
+class Note {
+    constructor(id, title, tags, body) {
+        this.id = id;
+        this.title = title;
+        this.tags = tags;
+        this.body = body;
+        this.pinToTop = false;
+        this.modified = getTime();
+    }
+}
+
+function storeState(state) {
+    localStorage.setItem("state", JSON.stringify(state));
+}
+
+export default function AppReducer(state, action) {
+    switch (action.type) {
```

Comment:
Suggestion. When using the Context API, you can try to separate it per concern. So in this case, there's gonna be three, NoteProvider, NavDrawerProvider, and TagProvider

You can check [this arcticle](https://dev.to/milu_franz/demystifying-react-hooks-usecontext-5g4a) as to why I mentioned that 😀